src/tests/system/tests/test_passkey.py

```diff
@@ -108,3 +108,140 @@ def test_passkey__su(client: Client, provider: GenericProvider, moduledatadir: s
         ioctl=f"{moduledatadir}/umockdev.ioctl",
         script=f"{testdatadir}/umockdev.script.{suffix}",
     )
+
+
+@pytest.mark.importance("high")
+@pytest.mark.topology(KnownTopology.LDAP)
+@pytest.mark.topology(KnownTopology.IPA)
+@pytest.mark.topology(KnownTopology.AD)
+@pytest.mark.topology(KnownTopology.Samba)
+@pytest.mark.builtwith(client="passkey", provider="passkey")
+def test_passkey__su_fail_pin(client: Client, provider: GenericProvider, moduledatadir: str, testdatadir: str):
+    """
+    :title: Check authentication deny of user with LDAP, IPA, AD and Samba with incorrect pin
+    :setup:
+        1. Add a LDAP, IPA, AD and Samba user with passkey_mapping.
+        2. Setup SSSD client with FIDO and umockdev, start SSSD service.
+    :steps:
+        1. Check authentication of the user with incorrect PIN.
+    :expectedresults:
+        1. User failed to authenticate.
+    :customerscenario: False
+    """
+    suffix = type(provider).__name__.lower()
+
+    if suffix == "ldap":
+        client.sssd.domain["local_auth_policy"] = "only"
+
+    with open(f"{testdatadir}/passkey-mapping.{suffix}") as f:
+        provider.user("user1").add().passkey_add(f.read().strip())
+
+    client.sssd.start()
+
+    assert not client.auth.su.passkey(
+        username="user1",
+        pin=67890,
+        device=f"{moduledatadir}/umockdev.device",
+        ioctl=f"{moduledatadir}/umockdev.ioctl",
+        script=f"{testdatadir}/umockdev.script.{suffix}",
+    )
+
+
+@pytest.mark.importance("high")
+@pytest.mark.topology(KnownTopology.LDAP)
+@pytest.mark.topology(KnownTopology.IPA)
+@pytest.mark.topology(KnownTopology.AD)
+@pytest.mark.topology(KnownTopology.Samba)
+@pytest.mark.builtwith(client="passkey", provider="passkey")
+def test_passkey__su_fail_mapping(client: Client, provider: GenericProvider, moduledatadir: str, testdatadir: str):
+    """
+    :title: Check authentication deny of user with LDAP, IPA, AD and Samba with incorrect mapping
+    :setup:
+        1. Add a LDAP, IPA, AD and Samba user with passkey_mapping.
+        2. Setup SSSD client with FIDO and umockdev, start SSSD service.
+    :steps:
+        1. Check authentication of the user with incorrect passkey mapping.
+    :expectedresults:
+        1. User failed to authenticate.
+    :customerscenario: False
+    """
+    suffix = type(provider).__name__.lower()
+
+    if suffix == "ldap":
+        client.sssd.domain["local_auth_policy"] = "only"
```

Comment:
Is this the standard pattern we should use for checking which Provider is in use?   I've used `isinstance(provider, LDAP)`.   This way, however, doesn't require importing the provider classes.   If this method is used elsewhere, that's great and I'll switch my own PR to use this way.   I just want to confirm this is how we want to start checking which provider is in use.

Reply:
`(Pdb) print(type(provider).__name__)`
`LDAP`

getting this but followings are not working,

`if isinstance(provider, (ldap)): print("anything")`
`*** NameError: name 'ldap' is not defined`

`if isinstance(provider, (LDAP)): print("anything")`
`*** NameError: name 'LDAP' is not defined`

is am I missing something?

Reply:
yes, it worked  but I again I need to declare the `suffix` also. So better can I keep as it.